### PR TITLE
refactor: Rewrite pipelines for better reusability

### PR DESCRIPTION
**Pull request by tumido**
Resolves: https://github.com/AICoE/meteor/issues/45

- Splits each pipeline blob into independent task resources
- Uses standardized catalogue `ClusterTask` resources wherever possible (container build and push) instead of custom logic
- Allows pipelines manifest to be a dead simple workflow definition
- Rewrites `base_image` analysis code for Notebook image to be much cleaner and smarter and also make it display much nicer output
- Disables Thoth image analyze for now (the code needs a rewrite before enabling it)

image

Pipelines depends on PVCs now, since Pipeline can't share an empty volume between Task in Tekton (unlike Task, which can share it between its steps)

### CODE REVIEW

File: config/pipelines/tasks/generate-jupyterhub.yaml [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        --env=THOTH_ADVISE=0 \
+        --env=THOTH_ERROR_FALLBACK=1 \
+        --env=THOTH_DRY_RUN=1 \
+        --env=THAMOS_DEBUG=0 \
+        --env=THAMOS_VERBOSE=1 \
+        --env=THOTH_PROVENANCE_CHECK=0 \
+        --env=GIT_REPO_URL=$(params.url) \
+        --env=GIT_REPO_NAME="$REPONAME"
+  #
+  # sidecars:
```
Comment (harshad16) [inline config/pipelines/tasks/generate-jupyterhub.yaml:148]: this has to be added to another task to feedback the log to project thoth

**Review by harshad16 — COMMENTED** [review]
/hold
kudos to the effort, really appreciate this
I see, all the tasks are been rewrite
if this tested can you let me know
i would need some time to check on this 

also, we had a feedback loop, which use to send images and logs for analysis
with  this new flow we should add them to separate tasks

**Comment by goern** [comment]
@tumido please create a #techDebt issue for this. all the tekton related declarations need to be part of AICoE-CI and should be refactored and provided via https://github.com/AICoE/aicoe-ci/issues/127

**Comment by tumido** [comment]
@goern The issue is here since Monday: https://github.com/AICoE/meteor/issues/90

File: config/pipelines/tasks/generate-jupyterhub.yaml (hunk 1 above)
Comment (tumido) [reply config/pipelines/tasks/generate-jupyterhub.yaml:237]: Yes. Once you're back from PTO let's sync on how thoth reporting is handled and what it actually means to send images back to Thoth etc. (If it makes sense in scenarios like when the image is private and stored in cluster only, on a different cluster than where Thoth is located)

Can we keep them off for now? Let's get the initial flow without thoth reporting and I'll work on that bit right after it. WDYT?

I've also added the `send-for-image-analysis` step back (commented out)

**Comment by tumido** [comment]
Let me self merge this so we can move on. We can always refactor it again.

**Pull request merged**